src/Sentry.Unity.Android/JniExecutor.cs

```diff
@@ -1,182 +1,58 @@
-                // We only care about the _taskEvent
-                break;
-            }
-
-            try
-            {
-                // Matching the type of the `_currentTask` exactly. The result gets cast to the expected type
-                // when returning from the blocking call.
-                switch (_currentTask)
-                {
-                    case Action action:
-                        {
-                            action.Invoke();
-                            _taskCompletionSource?.SetResult(null);
-                            break;
-                        }
-                    case Func<bool> func1:
-                        {
-                            var result = func1.Invoke();
-                            _taskCompletionSource?.SetResult(result);
-                            break;
-                        }
-                    case Func<bool?> func2:
-                        {
-                            var result = func2.Invoke();
-                            _taskCompletionSource?.SetResult(result);
-                            break;
-                        }
-                    case Func<string?> func3:
-                        {
-                            var result = func3.Invoke();
-                            _taskCompletionSource?.SetResult(result);
-                            break;
-                        }
-                    default:
-                        throw new NotImplementedException($"Task type '{_currentTask?.GetType()}' with value '{_currentTask}' is not implemented in the JniExecutor.");
-                }
-            }
-            catch (Exception e)
-            {
-                _logger?.LogError(e, "Error during JNI execution.");
-                _taskCompletionSource?.SetException(e);
-            }
+            _androidJNI.AttachCurrentThread();
         }
 
-        AndroidJNI.DetachCurrentThread();
-    }
-
-    public TResult? Run<TResult>(Func<TResult?> jniOperation, TimeSpan? timeout = null)
-    {
-        lock (_lock)
+        try
         {
-            timeout ??= DefaultTimeout;
-            using var timeoutCts = new CancellationTokenSource(timeout.Value);
-            _taskCompletionSource = new TaskCompletionSource<object?>();
-            _currentTask = jniOperation;
-            _taskEvent.Set();
-
-            try
-            {
-                _taskCompletionSource.Task.Wait(timeoutCts.Token);
-                return (TResult?)_taskCompletionSource.Task.Result;
-            }
-            catch (OperationCanceledException)
-            {
-                _logger?.LogError("JNI execution timed out.");
-                return default;
-            }
-            catch (Exception e)
-            {
-                _logger?.LogError(e, "Error during JNI execution.");
-                return default;
-            }
-            finally
-            {
-                _currentTask = null!;
-            }
+            return jniOperation.Invoke();
```

Comment:
We throw from this, right? e.g; when method doesn't exist? 
Where do we handle those exceptions?